h/search/query.py

```diff
@@ -15,6 +20,34 @@
 OFFSET_MAX = 9800
 
 
+def wildcard_uri_is_valid(wildcard_uri):
```

Comment:
This function returns false if the URL has no path component at all. As a result, in your follow up PR I encountered an error with `/api/search?wildcard_uri=http://example.org`. A wildcard URI with no actual wildcards in it is perhaps a bit pointless but I don't think it should be an error.

Reply:
I have fixed this but now it excepts ? inside the domain (not in the scheme just the netloc). I figure this is ok for now and if it causes slowdown issues we can fix it. I don't want to merge this without full-filing the basic requirements and I feel like this is a basic requirement.

Reply:
You've given this a thumbs up but this function still returns `False` if the URI does not contain any wildcards. Is that intentional?

Reply:
I thought you were saying it _should_ return False indicating it _is_ an invalid wildcard_uri which I think it should be? I would expect part of validating a wildcard uri is that it does contain wildcards, otherwise it's not a wildcard uri. That way we aren't handling both cases in one method (not that it should be a problem for elasticsearch results) but I don't think we should be using a wildcard search when there aren't any wildcards plus if the user is using our api incorrectly and passing in non-wildcards into the wildcard_uri it notifies them rather than letting them continue to use something incorrectly. Not to mention if the user selected wildcard_uri and didn't provide a wildcard uri, chances are they probably made a mistake and forgot to add a wildcard into the uri.